### PR TITLE
[next] Ensure middleware route comes before beforeFiles rewrites

### PR DESCRIPTION
**Pull request by ijjk**
This ensures `beforeFiles` rewrites come after middleware as there can be a conflict with redirects in middleware if they aren't expecting rewrites to modify the original path. 

### Related Issues

x-ref: https://github.com/vercel/vercel/pull/7847
x-ref: [slack thread](https://github.com/vercel/vercel/pull/7847)

#### Tests

- [ ] The code changed/added as part of this PR has been covered with tests
- [ ] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/next/test/fixtures/00-middleware/middleware.js [hunk 1]
```diff
@@ -47,6 +47,11 @@ export function middleware(request) {
     return;
   }
 
+  if (url.pathname === '/somewhere') {
```
Comment (styfle) [inline packages/next/test/fixtures/00-middleware/middleware.js:50]: How about a test to request the `/somewhere` path?

File: packages/next/src/server-build.ts [hunk 2]
```diff
@@ -1025,6 +1025,8 @@ export async function serverBuild({
 
       ...redirects,
 
+      ...middleware.staticRoutes,
```
Comment (EndangeredMassa) [inline packages/next/src/server-build.ts]: Should we add a comment here about why this needs to come before `beforeFilesRewrites`?

Comment (ijjk) [reply packages/next/src/server-build.ts]: Updated in https://github.com/vercel/vercel/pull/7912/commits/658bbb8187c5db20ed39549272661ab8df6bfa56

File: packages/next/test/fixtures/00-middleware/middleware.js (hunk 1 above)
Comment (ijjk) [reply packages/next/test/fixtures/00-middleware/middleware.js:50]: Added in https://github.com/vercel/vercel/pull/7912/commits/658bbb8187c5db20ed39549272661ab8df6bfa56

**Review by styfle — APPROVED** [review]
Thanks!

**Review by EndangeredMassa — APPROVED** [review]

**Pull request merged**